server/etcdmain/config_test.go

```diff
@@ -653,3 +654,101 @@ func validateClusteringFlags(t *testing.T, cfg *config) {
 		t.Errorf("advertise-client-urls = %v, want %v", cfg.ec.AdvertiseClientUrls, wcfg.ec.AdvertiseClientUrls)
 	}
 }
+
+func TestConfigFileDeprecatedOptions(t *testing.T) {
+	type configStruct struct {
+		SnapshotCount                           uint64        `json:"snapshot-count,omitempty"`
+		MaxSnapFiles                            uint          `json:"max-snapshots,omitempty"`
+		V2Deprecation                           string        `json:"v2-deprecation,omitempty"`
+		ExperimentalCompactHashCheckEnabled     bool          `json:"experimental-compact-hash-check-enabled,omitempty"`
+		ExperimentalCompactHashCheckTime        time.Duration `json:"experimental-compact-hash-check-time,omitempty"`
+		ExperimentalWarningUnaryRequestDuration time.Duration `json:"experimental-warning-unary-request-duration,omitempty"`
+	}
```

Comment:
I think that you can reuse [embed.Config](https://github.com/etcd-io/etcd/blob/9db8dcb44b9ad7d0cf94438169cef7a778e2eb15/server/embed/config.go#L162) directly instead of defining another struct.